packages/frontend/src/features/sqlRunner/hooks/useSqlQueryRun.tsx

```diff
@@ -0,0 +1,171 @@
+            },
+            onError: async () => {
+                await queryClient.cancelQueries([
+                    'jobStatus',
+                    sqlQueryJob?.jobId,
+                ]);
+            },
+            enabled: Boolean(sqlQueryJob && sqlQueryJob?.jobId !== undefined),
+        },
+    );
+
+    const { data: sqlQueryResults, isLoading: isResultsLoading } = useQuery<
+        ResultRow[] | undefined,
+        ApiError
+    >(
+        ['sqlQueryResults', sqlQueryJob?.jobId],
+        async () => {
+            const url = scheduledDeliveryJobStatus?.details?.fileUrl;
+            const response = await fetch(url, {
+                method: 'GET',
+                headers: {
+                    Accept: 'application/json',
+                },
+            });
+            const rb = response.body;
+            const reader = rb?.getReader();
+
+            const stream = new ReadableStream({
+                start(controller) {
+                    function push() {
+                        void reader?.read().then(({ done, value }) => {
+                            if (done) {
+                                // Close the stream
+                                controller.close();
+                                return;
+                            }
+                            // Enqueue the next data chunk into our target stream
+                            controller.enqueue(value);
+
+                            push();
+                        });
+                    }
+
+                    push();
+                },
+            });
+
+            const responseStream = new Response(stream, {
+                headers: { 'Content-Type': 'application/json' },
+            });
+            const result = await responseStream.text();
+
+            // Split the JSON strings by newline
+            const jsonStrings = result.trim().split('\n');
+            const jsonObjects = jsonStrings
+                .map((jsonString) => {
+                    try {
+                        return JSON.parse(jsonString);
+                    } catch (e) {
+                        throw new Error('Error parsing JSON');
+                    }
+                })
+                .filter((obj) => obj !== null);
+
+            return jsonObjects;
+        },
+        {
+            onError: () => {
+                showToastError({
+                    title: 'Could not fetch SQL query results',
+                });
+            },
+            enabled: Boolean(
+                scheduledDeliveryJobStatus?.status ===
+                    SchedulerJobStatus.COMPLETED &&
+                    scheduledDeliveryJobStatus?.details?.fileUrl !== undefined,
+            ),
+        },
+    );
+
+    const isLoading = useMemo(
+        () =>
+            scheduledDeliveryJobStatus?.status === SchedulerJobStatus.STARTED &&
+            isResultsLoading,
+        [scheduledDeliveryJobStatus?.status, isResultsLoading],
+    );
```

Comment:
should it have `sqlQueryRunMutation.isLoading` to account for the first query and for scheduledDeliveryJobStatus to be undefined?

Reply:
added! https://github.com/lightdash/lightdash/pull/10608/commits/b56c5f0c38eb520eeb5fab66f54870e988d07f62 and tested